pandas/io/parsers/python_parser.py

```diff
@@ -1166,6 +1164,26 @@ def _remove_skipped_rows(self, new_rows: list[list[Scalar]]) -> list[list[Scalar
             ]
         return new_rows
 
+    def _set_no_thousand_columns(self) -> set[int]:
+        no_thousands_columns: set[int] = set()
+        if self.columns and self.parse_dates:
+            assert self._col_indices is not None
+            no_thousands_columns = self._set_noconvert_dtype_columns(
+                self._col_indices, self.columns
+            )
+        if self.columns and self.dtype:
+            assert self._col_indices is not None
+            for i in self._col_indices:
+                if not isinstance(self.dtype, dict) and not is_numeric_dtype(
+                    self.dtype
+                ):
+                    no_thousands_columns.add(i)
+                if isinstance(self.dtype, dict) and not is_numeric_dtype(
+                    self.dtype.get(self.columns[i], None)
+                ):
+                    no_thousands_columns.add(i)
```

Comment:
sorry, I wasn't clear - what I meant was that if checking that `self.columns[i]` is in `self.dtype` is necessary to get something to pass, then that thing should be added as a test

So, let's add a test for which that check is necessary (`dtype={'a': str}` should be enough?) - it should work to just parametrise over the test you've added, so it runs with both `dtype` specifying all columns, and only specifying `'a'`?

Also, I think this can be simplified to something like
```python
                if (
                    isinstance(self.dtype, dict)
                    and self.columns[i] in self.dtype
                    and not is_numeric_dtype(self.dtype[self.columns[i]])
                ):
                    no_thousands_columns.add(i)
```

Reply:
> sorry, I wasn't clear - what I meant was that if checking that `self.columns[i]` is in `self.dtype` is necessary to get something to pass, then that thing should be added as a test
> 
> So, let's add a test for which that check is necessary (`dtype={'a': str}` should be enough?) - it should work to just parametrise over the test you've added, so it runs with both `dtype` specifying all columns, and only specifying `'a'`?
> 
Thank you, I did as you suggested. I added parametrization to the test, so it covers both cases.

> Also, I think this can be simplified to something like
> 
> ```python
>                 if (
>                     isinstance(self.dtype, dict)
>                     and self.columns[i] in self.dtype
>                     and not is_numeric_dtype(self.dtype[self.columns[i]])
>                 ):
>                     no_thousands_columns.add(i)
> ```
The simplification works like a charm :)